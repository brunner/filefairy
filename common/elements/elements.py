#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Common (non-reloadable) util methods for rendering templates."""

FAVICON_LINK = 'https://fairylab.surge.sh/favicon-32x32.png'


def _attr(attributes):
    if attributes is None:
        return ''

    items = sorted(attributes.items())
    return ' ' + ' '.join('{}="{}"'.format(k, v) for k, v in items)


def _trow(t, name, row):
    if row is None:
        return
    if name not in t or t[name] is None:
        t[name] = []
    t[name].append(row)


def cell(col=None, content=''):
    """Builds a cell object, describing a table cell.

    Args:
        col: The col object for this particular cell.
        content: The cell text.

    Returns:
        The cell object.
    """
    obj = {}
    if col:
        obj['col'] = col
    if content:
        obj['content'] = content
    return obj


def col(clazz='', colspan=''):
    """Builds a col object, describing a table column.

    Args:
        clazz: The CSS class(es) to apply to the column.
        colspan: The number of columns that this column should span.

    Returns:
        The col object.
    """
    obj = {}
    if clazz:
        obj['clazz'] = clazz
    if colspan:
        obj['colspan'] = colspan
    return obj


SITELINKS_HCOLS = [col(clazz='font-weight-bold text-dark')]
SITELINKS_BCOLS = [col(clazz='position-relative')]
SITELINKS_TABLES = [
    ('Tasks', [
        ('timer', '/gameday/', 'Gameday'),
        ('people', '/news/', 'News'),
        ('spreadsheet', '/standings/', 'Standings'),
    ]),
    ('Other', [
        ('dashboard', '/dashboard/', 'Dashboard'),
        ('home', '/', 'Home'),
    ]),
]


def anchor(url, content):
    """Builds an anchor element.

    Args:
        url: The link url.
        content: The inner HTML of the anchor element.

    Returns:
        The anchor element.
    """
    return '<a href="{}">{}</a>'.format(url, content)


def dialog(id_='', icon='', tables=None):
    """Builds a dialog object.

    Args:
        id_: The CSS id to set on the root dialog element.
        icon: The icon element to display in the dialog element header.
        tables: The optional list of table elements.

    Returns:
        The dialog object.
    """
    obj = {}
    if id_:
        obj['id'] = id_
    if icon:
        obj['icon'] = icon
    if tables:
        obj['tables'] = tables
    return obj


def icon_img(src, size, classes, id_):
    """Builds an icon element for an image resource.

    Args:
        src: The source of the image.
        size: The size of the image, in pixels.
        classes: The list of CSS classes to apply to the img.
        id_: The CSS id to apply to the img.

    Returns:
        The span element.
    """
    if id_:
        id_ = ' id="{}"'.format(id_)
    classes = ' class="{}"'.format(' '.join(classes)) if classes else ''
    tag = '<img src="{0}" width="{1}" height="{1}" border="0"{2}{3}>'
    return tag.format(src, size, id_, classes)


def icon_span(name='', attributes=None, classes=None):
    """Builds an icon element for an Iconic font span.

    Args:
        name: The name of the icon.
        attributes: The set of data attributes to apply to the span.
        classes: The list of CSS classes to apply to the span.

    Returns:
        The span element.
    """
    if classes is None:
        classes = []
    classes = ['oi', 'oi-' + name, 'absolute-icon'] + classes
    return span(classes=classes, attributes=attributes)


def menu():
    """Builds a menu dialog.

    Returns:
        The menu dialog.
    """
    img = icon_img(FAVICON_LINK, '16', ['absolute-icon', 'left'], '')
    span_ = span(classes=['d-block', 'px-4'], text='Fairylab')
    return dialog(id_='menu', icon=(img + span_), tables=sitelinks(home=True))


def pre(content):
    """Builds a pre element.

    Args:
        content: The inner HTML of the pre element.

    Returns:
        The pre element.
    """
    return '<pre>{}</pre>'.format(content)


def row(cells=None, attributes=None, clazz=''):
    """Builds a row object, describing a table row.

    Args:
        cells: The cells in the row.
        attributes: The set of data attributes to apply to the row.
        clazz: The CSS class(es) to apply to the row.

    Returns:
        The row object.
    """
    obj = {}
    if cells:
        obj['cells'] = cells
    if attributes:
        obj['attr'] = _attr(attributes)
    if clazz:
        obj['clazz'] = clazz
    return obj


def ruleset(selector='', rules=None):
    """Builds a CSS ruleset element.

    Args:
        selector: The CSS selector.
        rules. The list of rules.

    Returns:
        The ruleset element.
    """
    obj = {}
    if selector:
        obj['selector'] = selector
    if rules:
        obj['rules'] = rules
    return obj


def sitelinks(home=False):
    """Builds a list of sitelinks tables.

    Args:
        home: Whether or not to include a Home link in the sitelinks.

    Returns:
        The list of sitelinks tables.
    """
    tables = [topper('Site Links')]
    for head, rows in SITELINKS_TABLES:
        body = []
        for name, href, text in rows:
            if name == 'home' and not home:
                continue
            icon = icon_span(name=name, classes=['left', 'text-secondary'])
            span_ = span(classes=['d-block pl-4'], text=anchor(href, text))
            body.append(row(cells=[cell(content=(icon + span_))]))

        tables.append(
            table(
                clazz='border mb-3',
                hcols=SITELINKS_HCOLS,
                bcols=SITELINKS_BCOLS,
                head=[row(cells=[cell(content=head)])],
                body=body))

    return tables


def span(id_='', classes='', attributes=None, text=''):
    """Builds a span element.

    Args:
        id: The CSS id to apply to the span.
        classes: The list of CSS classes to apply to the span.
        attributes: The set of data attributes to apply to the span.
        content: The inner HTML of the span element.

    Returns:
        The span element.
    """
    attr = _attr(attributes)
    if id_:
        id_ = ' id="{}"'.format(id_)
    if classes:
        classes = ' class="{}"'.format(' '.join(classes))
    return '<span{}{}{}>{}</span>'.format(id_, classes, attr, text)


def table(clazz='',
          id_='',
          attributes=None,
          hcols=None,
          bcols=None,
          fcols=None,
          head=None,
          body=None,
          foot=None):
    """Builds a table element.

    Args:
        clazz: The optional CSS class to set on the root table element.
        id_: The optional CSS id to set on the root table element.
        attributes: The set of data attributes to apply to the table.
        hcols: The optional list of CSS classes to apply to each head column.
        bcols: The optional list of CSS classes to apply to each body column.
        fcols: The optional list of CSS classes to apply to each footer column.
        head: The optional list of head rows.
        body: The optional list of body rows.
        foot: The optional list of footer rows.

    Returns:
        The table element.
    """
    obj = {}
    if clazz:
        obj['clazz'] = clazz
    if id_:
        obj['id'] = id_
    if attributes:
        obj['attributes'] = _attr(attributes)
    if hcols:
        obj['hcols'] = hcols
    if bcols:
        obj['bcols'] = bcols
    if fcols:
        obj['fcols'] = fcols
    if head:
        obj['head'] = head
    if body:
        obj['body'] = body
    if foot:
        obj['foot'] = foot
    return obj


def tbody(obj, row):
    """Appends a body row to an existing table.

    Args:
        obj: The table.
        row: The row.
    """
    _trow(obj, 'body', row)


def topper(text):
    """Builds a topper table element.

    Args:
        text: The text of the topper table element.

    Returns:
        The topper table element.
    """
    return table(
        clazz='topper',
        bcols=[col(clazz='border-0 font-weight-bold px-0 text-secondary')],
        body=[row(cells=[cell(content=text)])])


def ts(half, minute, second):
    """Returns a string representing a livesim timestamp.

    Args:
        half: The index of the inning.
        minute: The index of the minute.
        second: The index of the second.

    Returns:
        The timestamp string.
    """
    return '{}.{}.{}'.format(half, minute, second)
