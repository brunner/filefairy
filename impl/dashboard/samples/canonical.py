#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Sample data for dashboard.py golden test."""

import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
sys.path.append(re.sub(r'/impl/dashboard/samples', '', _path))

from common.elements.elements import anchor  # noqa
from common.elements.elements import cell  # noqa
from common.elements.elements import col  # noqa
from common.elements.elements import dialog  # noqa
from common.elements.elements import pre  # noqa
from common.elements.elements import row  # noqa
from common.elements.elements import table  # noqa

LINK = 'https://github.com/brunner/filefairy/blob/master/'

_error_title = 'filefairy.py#L127'
_error_link = LINK + 'impl/filefairy/' + _error_title
_error_msg = 'Disabled snacks.'

_exc = """Traceback (most recent call last):
  File "impl/filefairy/filefairy.py", line 127, in _foo
    a = func(*args, **kwargs)
  File "/home/pi/filefairy/tasks/snacks/snacks.py", line 234, in _bar
    b = self._baz()
  File "/home/pi/filefairy/tasks/snacks/snacks.py", line 190, in _baz
    return variable
NameError: name 'variable' is not defined
"""

_info_title = 'leaguefile.py#L317'
_info_link = LINK + 'tasks/leaguefile/' + _info_title
_info_msg = 'File is up.'

_logs = [
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark', colspan='3')],
        bcols=[col(clazz='w-150p'), None,
               col(clazz='text-right w-75p')],
        fcols=[col(colspan='3')],
        head=[row(cells=[cell(content='Saturday, October 26th, 1985')])],
        body=[
            row(cells=[
                cell(content=anchor(_error_link, _error_title)),
                cell(content=_error_msg),
                cell(content='06:02')
            ])
        ],
        foot=[row(cells=[cell(content=pre(_exc))])]),
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark', colspan='3')],
        bcols=[col(clazz='w-150p'), None,
               col(clazz='text-right w-75p')],
        fcols=[col(colspan='3')],
        head=[row(cells=[cell(content='Saturday, October 26th, 1985')])],
        body=[
            row(cells=[
                cell(content=anchor(_info_link, _info_title)),
                cell(content=_info_msg),
                cell(content='02:30')
            ]),
        ]),
]

subtitle = ''

tmpl = 'dashboard.html'

context = {
    'logs': _logs,
}
