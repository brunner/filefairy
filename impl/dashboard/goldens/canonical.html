<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<meta name="msapplication-TileColor" content="#dfe3e7">
<meta name="theme-color" content="#dfe3e7">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab Â· Dashboard</title>

<style>
html {
  min-height: 100%;
  overflow-y: scroll;
  position: relative;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
pre {
  margin: 0;
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
}
a.disabled{
  pointer-events: none;
  cursor: default;
}
.absolute-icon {
  position: absolute;
  top: 13px;
}
span.absolute-icon {
  border-radius: .25rem;
  padding: 3px;
  top: 12px;
}
.absolute-icon.left {
  left: 10px;
}
.absolute-icon.right {
  left: auto;
  right: 10px;
}
.absolute-icon.hover:hover {
  background-color: #e9ecef;
  cursor: pointer;
}
.alert, .dropdown-toggle, .form-control {
  padding: .75rem;
}
.alert .oi {
  line-height: 24px;
  padding-right: .75rem;
}
.alert-danger {
  border: 1px solid #721c24;
}
.alert-primary {
  border: 1px solid #004085;
}
.alert-success {
  border: 1px solid #155724;
}
.anchor {
  padding-top: 0px;
  margin-top: 0px;
  transition: padding-top .35s ease, margin-top .35s ease;
}
.badge-icon {
  border: 1px solid #dee2e6;
  cursor: pointer;
  font-weight: 400!important;
  font-size: 100%;
  padding: .25rem .5rem;
  white-space: normal;
  width: 100%;
}
.badge-icon[data-dismiss="modal"]:hover,
.badge-icon[data-toggle="modal"]:hover {
  background-color: #e9ecef;
  border-color: #dee2e6;
}
.before-score {
  padding-right: .75rem;
}
button.close {
  height: 16px;
  line-height: 16px;
  width: 16px;
}
.carousel-fade .carousel-item {
  opacity: 0;
  transition-duration: .6s;
  transition-property: opacity;
}
.carousel-fade .carousel-item.active,
.carousel-fade .carousel-item-next.carousel-item-left,
.carousel-fade .carousel-item-prev.carousel-item-right {
  opacity: 1;
}
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-right {
  opacity: 0;
}
.carousel-fade .carousel-item-next,
.carousel-fade .carousel-item-prev,
.carousel-fade .carousel-item.active,
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-prev {
  transform: translateX(0);
  transform: translate3d(0, 0, 0);
}
.dropdown-toggle {
 color: #6c767c;
}
.flex-grow {
  flex-grow: 1;
}
.grayscale {
  -webkit-filter: grayscale(1) opacity(0.5);
  filter: grayscale(1) opacity(0.5);
}
.head {
  border-left: 1px solid transparent;
  border-right: 1px solid transparent;
  padding: .75rem;
}
.h-58p {
  height: 58px;
}
.h-72p {
  height: 72px;
}
.modal {
  overflow-y: auto;
}
.modal-header {
  background-color: #f5f5f5;
  border-bottom: 1px solid #dee2e6;
}
.modal-open {
  overflow: auto;
  padding-right: 0px !important;
}
.mr-8p {
  margin-right: 8px;
}
.nav-pills {
  overflow-x: auto;
  overflow-y: hidden;
}
.navbar {
  padding: 0;
}
.nav-pills .nav-item {
  margin: 0 .25rem;
  min-width: 32px;
  text-align: center;
}
.nav-pills .nav-link {
  padding: .25rem 0;
}
.nav-pills .nav-link:not(.active) {
  border: 1px solid #dee2e6;
}
.page-link {
  background-color: #f8f9fa;
  line-height: 1;
  padding: .25rem .75rem;
}
.page-link-inner {
  color: #212529;
  display: inline-block;
  padding: 0;
  padding-top: 1px;
}
.pagination {
  margin: 0;
  position: absolute;
  right: 0;
  top: 4px;
}
.pitch {
  border-radius: 16px;
  height: 16px;
  line-height: 16px;
  margin-left: 15px;
  margin-right: 26px;
  padding: 0 .4em;
  text-align: center;
  width: 24px;
  vertical-align: top;
}
.profile-text {
  min-height: 72px;
  padding-left: 65px;
}
.rt-10p {
  right: 10px;
  top: 10px;
}
.right-8p {
  right: 8px;
}
.table {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.small.table td {
  padding: .75rem .5rem;
}
.small.table td:first-child {
  padding-left: .75rem;
}
.small.table td:last-child {
  padding-right: .75rem;
}
.small.table td.badge-icon-wrapper {
  padding: .5rem .25rem;
}
.small.topper {
  font-size: 90%;
}
.table.collapse.show {
  display: table;
}
.tag {
  background-color: #ffffff;
  line-height: 16px;
  padding: 0 .75em;
  vertical-align: top;
}
.tag-light {
  border-color: #1b1e21!important;
  color: #1b1e21;
}
.text-underline {
  text-decoration: underline;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-50 {
  width: 50%;
}
.w-24p {
  width: 24px;
}
.w-28p {
  width: 28px;
}
.w-32p {
  width: 32px;
}
.w-40p {
  width: 40px;
}
.w-50p {
  width: 50px;
}
.w-65p {
  width: 65px;
}
.w-75p {
  width: 75px;
}
.w-95p {
  width: 95px;
}
.w-105p {
  width: 105px;
}
.w-150p {
  width: 150px;
}
.badge-icon-button {
  display: block;
  padding: 3px 0 2px;
}
.badge-icon-image,
.badge-icon-text {
  display: block;
}
.badge-icon-image {
  margin: 1px auto 0;
}
.badge-icon-text {
  color: #212529;
  padding: 0;
  padding-top: .25rem;
}
.td-sm-none,
.td-md-none,
.td-lg-none {
  display: none;
}
.td-md-show {
  display: table-cell;
}
@media (min-width: 576px) {
  body.modal-open header {
    margin-right: 0!important;
    padding-right: 0!important;
  }
  .badge-icon-image,
  .badge-icon-text {
    display: inline-block;
  }
  .badge-icon-text {
    padding: 2px 0 2px .5rem;
  }
  .td-sm-none {
    display: table-cell;
  }
}
@media (min-width: 768px) {
  .badge-icon-image,
  .badge-icon-text {
    display: block;
  }
  .badge-icon-text {
    padding: 0;
    padding-top: .25rem;
  }
  .modal-dialog {
    max-width: 650px; 
  }
  .td-md-none {
    display: table-cell;
  }
  .td-md-show {
    display: none
  }
}
@media (min-width: 992px) {
  .badge-icon-image,
  .badge-icon-text {
    display: inline-block;
  }
  .badge-icon-text {
    padding: 2px 0 2px .5rem;
  }
  .modal-dialog {
    max-width: 800px; 
  }
  .td-lg-none {
    display: table-cell;
  }
}
.count {
  line-height: 8px;
}
.dot {
  background-color: #ffffff;
  border-radius: 8px;
  height: 8px;
  margin: 2px 2px 2px 0;
  width: 8px;
  display: inline-block;
}
.dot.ball.active {
  border-color: #155724!important;
  background-color: #d4edda;
}
.dot.strike.active {
  border-color: #721c24!important;
  background-color: #f8d7da;
}
.tag {
  background-color: #ffffff;
  line-height: 16px;
}
.tag-danger {
  background-color: #f8d7da;
  border-color: #721c24!important;
  color: #721c24;
}
.tag-primary {
  background-color: #cce5ff;
  border-color: #004085!important;
  color: #004085;
}
.tag-secondary {
  background-color: #e2e3e5;
  border-color: #383d41!important;
  color: #383d41;
}
.tag-success {
  background-color: #d4edda;
  border-color: #155724!important;
  color: #155724;
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<div class="head position-relative small text-secondary">
<img src="https://fairylab.surge.sh/favicon-32x32.png"
     width="16" height="16" border="0" class="absolute-icon left">
<div class="px-4">Dashboard &nbsp;|&nbsp; 06:02:30 PDT (1985-10-26)</div>
<span class="oi oi-menu absolute-icon hover right"
      data-toggle="modal" data-target="#menu"></span></div>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pt-3">

<!-- ^(dashboard.logs) -->
<div class="container">

<!-- ^(common.topper) -->
<div class="row">
<div class="col-12">

<!-- ^(common.topper.table) -->
<table class="small table topper">

<!-- ^(common.topper.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Log Records</td>
</tr>
</tbody>
<!-- $(common.topper.body) -->

</table>
<!-- $(common.topper.table) -->

</div>
</div>
<!-- $(common.topper) -->

<!-- ^(dashboard.row) -->
<div class="row">

<!-- ^(dashboard.column) -->
<div class="col-12">

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark" colspan="3">Saturday, October 26th, 1985</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="w-150p"><a href="https://github.com/brunner/filefairy/blob/master/impl/filefairy/filefairy.py#L127">filefairy.py#L127</a></td>
<td>Disabled snacks.</td>
<td class="text-right w-75p">06:02</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

<!-- ^(common.table.foot) -->
<tfoot class="thead-light">
<tr>
<td scope="col" colspan="3"><pre>Traceback (most recent call last):
  File "impl/filefairy/filefairy.py", line 127, in _foo
    a = func(*args, **kwargs)
  File "/home/pi/filefairy/tasks/snacks/snacks.py", line 234, in _bar
    b = self._baz()
  File "/home/pi/filefairy/tasks/snacks/snacks.py", line 190, in _baz
    return variable
NameError: name 'variable' is not defined
</pre></td>
</tr>
</tfoot>
<!-- $(common.table.foot) -->

</table>
<!-- $(common.table) -->
<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark" colspan="3">Saturday, October 26th, 1985</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="w-150p"><a href="https://github.com/brunner/filefairy/blob/master/tasks/leaguefile/leaguefile.py#L317">leaguefile.py#L317</a></td>
<td>File is up.</td>
<td class="text-right w-75p">02:30</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(dashboard.column) -->

</div>
<!-- $(dashboard.row) -->

</div>
<!-- $(dashboard.logs) -->

<!-- ^(common.dialog) -->
<div class="modal fade" id="menu" tabindex="-1" role="dialog"
     aria-labelledby="menu-label" aria-hidden="true">
<div class="modal-dialog" role="document">
<div class="modal-content">

<!-- ^(common.dialog.header) -->
<div class="modal-header d-block py-0">
<div class="head position-relative small text-secondary">
<img src="https://fairylab.surge.sh/favicon-32x32.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block px-4">Fairylab</span>
<button type="button" class="close absolute-icon right m-0 p-0"
        data-dismiss="modal" aria-label="Close">
<span aria-hidden="true">&times;</span>
</button>
</div>
</div>
<!-- $(common.dialog.header) -->

<!-- ^(common.dialog.body) -->
<div class="modal-body pb-0">
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Site Links</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Tasks</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-timer absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/gameday/">Gameday</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-people absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/news/">News</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-spreadsheet absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/standings/">Standings</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Other</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-dashboard absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/dashboard/">Dashboard</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-home absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/">Home</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.dialog.body) -->

</div>
</div>
</div>
<!-- $(common.dialog) --></main>
<!-- $(base.main) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>


</body>

</html>